rascal-lsp/src/test/java/org/rascalmpl/vscode/lsp/rascal/model/PathConfigDiagnosticsTest.java

```diff
@@ -0,0 +1,171 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.rascalmpl.vscode.lsp.rascal.model;
+
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.PublishDiagnosticsParams;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.junit.Before;
+import org.junit.Test;
+import org.rascalmpl.library.Messages;
+import org.rascalmpl.library.Prelude;
+import org.rascalmpl.uri.URIResolverRegistry;
+import org.rascalmpl.uri.URIUtil;
+import org.rascalmpl.values.IRascalValueFactory;
+import org.rascalmpl.vscode.lsp.util.locations.ColumnMaps;
+
+import io.usethesource.vallang.IConstructor;
+import io.usethesource.vallang.ISourceLocation;
+
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+
+public class PathConfigDiagnosticsTest {
+    private static final IRascalValueFactory VF = IRascalValueFactory.getInstance();
+
+    private static final String PROJECT_A = "project-a";
+    private static final String PROJECT_B = "project-b";
+    private static final String POM_A = "project-a/pom.xml";
+    private static final String POM_B = "project-b/pom.xml";
+
+    private static final String MSG1 = "Hello World 1!";
+    private static final String MSG2 = "Hello World 2!";
+
+    private LanguageClient mockedClient;
+    private PathConfigDiagnostics sut;
+    private ColumnMaps columnMaps;
+
+    private final ISourceLocation projectA;
+    private final ISourceLocation projectB;
+
+    private final String pomAUrl;
+    private final String pomBUrl;
+
+    public PathConfigDiagnosticsTest() {
+        projectA = resourceLocation(PROJECT_A);
+        projectB = resourceLocation(PROJECT_B);
+
+        pomAUrl = resourceLocation(POM_A).getURI().toString();
+        pomBUrl = resourceLocation(POM_B).getURI().toString();
+    }
+
+    @Before
+    public void before() throws URISyntaxException {
+        mockedClient = mock(LanguageClient.class);
+        columnMaps = new ColumnMaps(this::getContents);
+        sut = new PathConfigDiagnostics(mockedClient, columnMaps);
+    }
+
+    private String resourceUrl(String res) {
```

Comment:
nit: I think this should return a URI, leave it to the consumers to downgrade it to a string.